Source/WebCore/workers/shared/context/SharedWorkerContextManager.cpp

```diff
@@ -57,7 +57,10 @@ void SharedWorkerContextManager::stopSharedWorker(SharedWorkerIdentifier sharedW
     // FIXME: We should be able to deal with the thread being unresponsive here.
 
     auto& thread = worker->thread();
-    thread.stop([worker = WTFMove(worker)]() mutable {
+    thread.stop([worker = WTFMove(worker), sharedWorkerIdentifier]() mutable {
+        if (auto* connection = SharedWorkerContextManager::singleton().connection())
+            connection->sharedWorkerTerminated(sharedWorkerIdentifier);
```

Comment:
Do we need to do it here?
It seems we should call sharedWorkerTerminated as soon as we remove the worker from the map?

Reply:
I matched the corresponding service worker code iirc but you may be right. I'll check.